server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -240,6 +246,17 @@ public TcpTransport(String transportName, Settings settings, ThreadPool threadPo
         this.networkService = networkService;
         this.transportName = transportName;
         defaultConnectionProfile = buildDefaultConnectionProfile(settings);
+        final Settings defaultFeatures = DEFAULT_FEATURES_SETTING.get(settings);
+        if (defaultFeatures == null) {
+            this.features = new String[0];
+        } else {
+            defaultFeatures.names().forEach(key -> {
+                if (Booleans.parseBoolean(defaultFeatures.get(key)) == false) {
+                    throw new IllegalArgumentException("feature settings must have default [true] value");
+                }
+            });
+            this.features = new TreeSet<>(defaultFeatures.names()).toArray(new String[defaultFeatures.names().size()]);
```

Comment:
can you add a comment that the goal of the TreeSet here is to bring the features into consistent order?

Reply:
I pushed 6d529fee20a88a87592337e9a49fd24a3bb06b05.